processor/groupbytraceprocessor/processor.go

```diff
@@ -233,7 +233,14 @@ func (sp *groupByTraceProcessor) onTraceReleased(rss []pdata.ResourceSpans) erro
 	}
 	stats.Record(context.Background(), mReleasedSpans.M(int64(trace.SpanCount())))
 	stats.Record(context.Background(), mReleasedTraces.M(1))
-	return sp.nextConsumer.ConsumeTraces(context.Background(), trace)
+
+	// Do async consuming not to block event worker
+	go func() {
```

Comment:
Can you add a test that demonstrates that this happens in a go routine?

Reply:
Imo this is too excessive to test 5-line function. There is a serious bug reason though, so I'll do that.